### PR TITLE
Fix padding for conv and pool operations.

### PR DESCRIPTION
**Pull request by ebarsoum**
Based on our last meeting, here the updated pads semantic.

### CODE REVIEW

File: docs/Operators.md [hunk 1]
```diff
@@ -687,12 +693,14 @@
...
     <dl>
+      <dt>auto_pad</dt>
+      <dd>If specified, it will overrides pads attribute values.auto_pad takes VALID or SAME, VALID means don't pad and same meanspad by matching the input spatial size.</dd>
       <dt>dilations</dt>
       <dd>Dilation along each axis, 1 means no dilation.</dd>
       <dt>kernel_shape</dt>
       <dd>The size of the kernel along each axis.</dd>
       <dt>pads</dt>
-      <dd>Padding along each axis, can take the value 0 (False) or non 0 (True)</dd>
+      <dd>Padding along each axis, can take any value greater than or equal to 0.The number represent the added pixels in the lower and upper part of thecorresponding  axis.</dd>
```
Comment (bddppq) [inline docs/Operators.md]: Typo: "thecorresponding"

Could you also clearly state that the added pixels will be padded as 0?

File: onnx/defs/nn/defs.cc [hunk 2]
```diff
@@ -98,7 +112,7 @@ The convolution operator consumes an input tensor and {filter_desc}, and
 computes the output.)DOC";
             ReplaceAll(doc, "{filter_desc}", filter_desc);
             schema.SetDoc(doc);
-            schema.NumInputs(2, 3);
```
Comment (bddppq) [inline onnx/defs/nn/defs.cc]: I agree with this change, but please point it out in the PR, it's so easy to overlook.

File: onnx/defs/nn/defs.cc [hunk 3]
```diff
@@ -181,8 +202,15 @@ and computes the output.)DOC";
             schema.Attr("strides",
                         "stride along each axis.",
                         AttrType::INTS);
+            schema.Attr("auto_pad",
+                        "If specified, it will overrides pads attribute values."
+                        "auto_pad takes VALID or SAME, VALID means don't pad and same means"
```
Comment (bddppq) [inline onnx/defs/nn/defs.cc]: "and SAME"

File: onnx/defs/nn/defs.cc [hunk 4]
```diff
@@ -127,8 +141,15 @@ computes the output.)DOC";
             schema.Attr("strides",
                         "stride along each axis.",
                         AttrType::INTS);
+            schema.Attr("auto_pad",
```
Comment (bddppq) [inline onnx/defs/nn/defs.cc]: Do we want to bikeshed the attribute name here? :-) @ezyang @dzhulgakov

**Comment by bddppq** [comment]
Overall LGTM

File: onnx/defs/nn/defs.cc (hunk 4 above)
Comment (ebarsoum) [reply onnx/defs/nn/defs.cc]: I don't care about the name, it just need to be different than pads.  So let me know which name you guys prefer.

**Comment by fmassa** [comment]
Do you plan to follow the same semantics of TF for the SAME case? Because it's not equivalent to just add `pads = (kernel_size - 1 )/ 2`, but it depends instead on the input size being even or odd as well.

**Comment by ebarsoum** [comment]
@fmassa that is a good question, it primarily depends if the resulted computation is symmetric or non-symmetric. @Yangqing and @yuanbyu any opinion on this part? Should the behavior be similar to TF (append the extra pixel to the end part of the dimension)?

**Comment by dzhulgakov** [comment]
I think the point of adding SAME is exactly to emulate TF behavior. Otherwise we could have expressed it with `pads` property

**Comment by ebarsoum** [comment]
@dzhulgakov not exactly, the main reason is to cover most cases easily.

**Pull request closed without merging**